File: Programación Orientada a Objetos/Clase6/Actividad2/src/Cafetera.java

public class Cafetera {
    private int capacidadMaxima;
    private int cantidadActual;

    // Constructor predeterminado
    public Cafetera() {
        capacidadMaxima = 1000;
        cantidadActual = 0;
    }

    // Constructor con la capacidad máxima de la cafetera
    public Cafetera(int capacidadMaxima) {
        this.capacidadMaxima = capacidadMaxima;
        cantidadActual = capacidadMaxima;
    }

    // Constructor con la capacidad máxima y la cantidad actual
    public Cafetera(int capacidadMaxima, int cantidadActual) {
        this.capacidadMaxima = capacidadMaxima;
        if (cantidadActual > capacidadMaxima) {
            this.cantidadActual = capacidadMaxima;
        } else {
            this.cantidadActual = cantidadActual;
        }
    }

    // Llena la cafetera
    public void llenarCafetera() {
        cantidadActual = capacidadMaxima;
    }

    // Sirve una taza con la capacidad indicada
    public void servirTaza(int capacidadTaza) {
        if (cantidadActual >= capacidadTaza) {
            cantidadActual -= capacidadTaza;
            System.out.println("Se ha servido una taza de " + capacidadTaza + " c.c. de café.");
        } else {
            System.out.println("No hay suficiente café para llenar la taza.");
            cantidadActual = 0;
        }
    }

    // Vacía la cafetera
    public void vaciarCafetera() {
        cantidadActual = 0;
    }

    // Agrega la cantidad de café indicada a la cafetera
    public void agregarCafe(int cantidad) {
        if (cantidadActual + cantidad <= capacidadMaxima) {
            cantidadActual += cantidad;
            System.out.println("Se han agregado " + cantidad + " c.c. de café a la cafetera.");
        } else {
            cantidadActual = capacidadMaxima;
            System.out.println("La cafetera está llena. No se puede agregar más café.");
        }
    }

    // Getter para obtener la cantidad actual de café en la cafetera
    public int getCantidadActual() {
        return cantidadActual;
    }

    // Getter para obtener la capacidad máxima de la cafetera
    public int getCapacidadMaxima() {
        return capacidadMaxima;
    }
}

File: Programación Visual/Clase11/Cafetera/Cafetera/Cafetera.vb
Public Class Cafetera
    Private capacidadMaxima As Integer
    Private cantidadActual As Integer

    ' Constructor
    Public Sub New(capacidadMaxima As Integer)
        Me.capacidadMaxima = capacidadMaxima
        Me.cantidadActual = 0
    End Sub

    ' Método para llenar la cafetera
    Public Sub LlenarCafetera()
        cantidadActual = capacidadMaxima
    End Sub

    ' Método para servir una taza
    Public Sub ServirTaza(cantidad As Integer)
        If cantidadActual >= cantidad Then
            cantidadActual -= cantidad
            Console.WriteLine("Se ha servido una taza de " & cantidad & " ml de café.")
        Else
            Console.WriteLine("No hay suficiente café para llenar la taza. Se ha servido lo que quedaba.")
            cantidadActual = 0
        End If
    End Sub

    ' Método para vaciar la cafetera
    Public Sub VaciarCafetera()
        cantidadActual = 0
        Console.WriteLine("La cafetera ha sido vaciada.")
    End Sub

    ' Método para agregar café a la cafetera
    Public Sub AgregarCafe(cantidad As Integer)
        If cantidadActual + cantidad <= capacidadMaxima Then
            cantidadActual += cantidad
            Console.WriteLine("Se ha agregado " & cantidad & " ml de café a la cafetera.")
        Else
            cantidadActual = capacidadMaxima
            Console.WriteLine("La cafetera está llena. No se puede agregar más café.")
        End If
    End Sub

    ' Método para obtener la cantidad actual de café en la cafetera
    Public Function ObtenerCantidadActual() As Integer
        Return cantidadActual
    End Function

    ' Método para obtener la capacidad máxima de la cafetera
    Public Function ObtenerCapacidadMaxima() As Integer
        Return capacidadMaxima
    End Function
End Class
